packages/react-native-editor/src/test/index.test.js

```diff
@@ -27,158 +29,202 @@ const initGutenberg = ( registerParams ) => {
 
-			return {
+			// An empty component is provided in order to listen for render calls of the editor component.
+			const onRenderEditor = jest.fn();
+			const EditorComponent = () => {
+				onRenderEditor();
+				return null;
+			};
+			jest.mock( '../setup', () => ( {
 				__esModule: true,
-				default: jest.fn().mockReturnValue( <></> ),
+				default: jest.fn().mockReturnValue( <EditorComponent /> ),
+			} ) );
+
+			initGutenberg();
+
+			const hookCallIndex = 0;
+			// "invocationCallOrder" can be used to compare call orders between different mocks.
+			// Reference: https://git.io/JyBk0
+			const hookCallOrder =
+				applyFilters.mock.invocationCallOrder[ hookCallIndex ];
+			const onRenderEditorCallOrder =
+				onRenderEditor.mock.invocationCallOrder[ 0 ];
+			const hookName = applyFilters.mock.calls[ hookCallIndex ][ 0 ];
+
+			expect( hookName ).toBe( 'native.block_editor_props' );
+			expect( hookCallOrder ).toBeLessThan( onRenderEditorCallOrder );
+		} );
+
+		it( 'dispatches "native.render" hook after the editor is rendered', () => {
+			const doAction = jest.spyOn( wpHooks, 'doAction' );
+
+			// An empty component is provided in order to listen for render calls of the editor component.
+			const onRenderEditor = jest.fn();
+			const EditorComponent = () => {
+				onRenderEditor();
+				return null;
 			};
+			jest.mock( '../setup', () => ( {
+				__esModule: true,
+				default: jest.fn().mockReturnValue( <EditorComponent /> ),
+			} ) );
+
+			initGutenberg();
+
+			const hookCallIndex = 1;
+			// "invocationCallOrder" can be used to compare call orders between different mocks.
+			// Reference: https://git.io/JyBk0
+			const hookCallOrder =
+				doAction.mock.invocationCallOrder[ hookCallIndex ];
+			const onRenderEditorCallOrder =
+				onRenderEditor.mock.invocationCallOrder[ 0 ];
+			const hookName = doAction.mock.calls[ hookCallIndex ][ 0 ];
+
+			expect( hookName ).toBe( 'native.render' );
+			expect( hookCallOrder ).toBeGreaterThan( onRenderEditorCallOrder );
 		} );
+	} );
 
-		initGutenberg( { beforeInitCallback } );
+	describe( 'editor initialization', () => {
+		beforeEach( () => {
+			// Setup already registers blocks so we need assure that no blocks are registered before the test.
+			getBlockTypes().forEach( ( block ) => {
+				unregisterBlockType( block.name );
+			} );
+		} );
 
-		// "invocationCallOrder" can be used to compare call orders between different mocks.
-		// Reference: https://git.io/JyBk0
-		const beforeInitCallOrder =
-			beforeInitCallback.mock.invocationCallOrder[ 0 ];
-		const onSetupImportedCallOrder =
-			mockOnModuleImported.mock.invocationCallOrder[ 0 ];
+		it( 'initializes the editor with empty HTML', async () => {
+			const { getByTestId } = initGutenberg( {}, { initialData: '' } );
 
-		expect( beforeInitCallOrder ).toBeLessThan( onSetupImportedCallOrder );
-	} );
+			const blockList = await waitFor( () =>
+				getByTestId( 'block-list-wrapper' )
+			);
```

Comment:
The usage of `waitFor` does not appear to be necessary. A synchronous query works. WDYT about avoiding asynchronous queries unless they are required? We could also remove the `async` decorator on the anonymous test function. 

```js
const blockList = getByTestId( 'block-list-wrapper' );
```

Reply:
Right, I agree with only providing asynchronous queries when required. I'll update all cases with your suggestion, thanks 🙇 !

Reply:
I applied this suggestion in https://github.com/WordPress/gutenberg/pull/37955/commits/b1964b5eedef8c3f5667947be81211189cc34a76.